PHC-Android/PHC/src/main/java/phc/android/Checkin/PersonalInfoFragment.java

```diff
@@ -311,4 +314,78 @@ private void prepopulateForm() {
         mMilitaryCheckbox.setChecked(military);
         mFosterCheckbox.setChecked(foster);
     }
+
+    private class OnPersonalInfoContinueClickListener extends OnContinueClickListener {
+
+        public OnPersonalInfoContinueClickListener(Context context, Fragment currFrag, ViewGroup currLayout, Fragment nextFrag, String nextFragName) {
+            super(context, currFrag, currLayout, nextFrag, nextFragName);
+        }
+
+        /**
+         * When continue button is clicked, updates SharedPreferences and loads the next fragment.
+         * SelectServicesFragment is handled by a different method because it has dynamically
+         * added checkbox views.
+         */
+        @Override
+        public void onClick(View view) {
+            if (!validateFields()) {
+                return;
+            }
+            super.onClick(view);
+        }
+    }
+
+    /**
+     * Corrects fields within the form and does a validation check
+     * @return false if some fields need correcting, true if okay
+     */
+    private boolean validateFields() {
+        final String monthText = mMonth.getText().toString();
+        final String dayText = mDay.getText().toString();
+        final String yearText = mYear.getText().toString();
+        final String emailText = mEmail.getText().toString();
+
+        if (!(yearText.length() == 4 || yearText.length() == 0)) {
+            Toast.makeText(getActivity(),
+                    "Please enter in 4 digits for the birthday year field",
+                    Toast.LENGTH_LONG).show();
+            return false;
+        }
+
+        if (!(monthText.matches("^[0-9]*$") &&
+                dayText.matches("^[0-9]*$") &&
+                yearText.matches("^[0-9]*$"))) {
+            Toast.makeText(getActivity(),
+                    "Please only enter in numerical digits for the birthday fields",
+                    Toast.LENGTH_LONG).show();
+            return false;
+        }
```

Comment:
should we consider also checking to see if the date is a valid date as a tighter validation?
http://stackoverflow.com/questions/11480542/fastest-way-to-tell-if-a-string-is-a-valid-date